java/component/keyple-core/src/main/java/org/eclipse/keyple/core/seproxy/plugin/reader/AbstractLocalReader.java

```diff
@@ -458,323 +324,301 @@ public void setSeProtocolSetting(Map<SeProtocol, String> protocolSetting) {
    * @param protocolFlag the protocol flag
    * @return true if the current protocol matches the provided protocol flag
    * @throws KeypleReaderIOException if the communication with the reader or the SE has failed
+   * @since 0.9
    */
   protected abstract boolean protocolFlagMatches(SeProtocol protocolFlag);
 
-  /* ==== SeRequestSe and SeRequest transmission management ============= */
+  /**
+   * Executes the ChannelControl instruction.
+   *
+   * <p>Does nothing if channelControl is {@link ChannelControl#KEEP_OPEN}<br>
+   * Otherwise, closes the logical channel and initiates the closure of the physical channel.
+   * Instantly if the reader is not observed or by the removal sequence if it is.
+   *
+   * @param channelControl The channel control.
+   */
+  private void processChannelControl(ChannelControl channelControl) {
+
+    if (channelControl != ChannelControl.KEEP_OPEN) {
+      // close logical channel unconditionally
+      closeLogicalChannel();
+      if (!(this instanceof ObservableReader)
+          || (((ObservableReader) this).countObservers() == 0)) {
+        /* Not observable/observed: close immediately the physical channel if requested */
+        closePhysicalChannel();
+      }
+      if (this instanceof AbstractObservableLocalReader) {
+        /*
+         * request the removal sequence when the reader is monitored by a thread
+         */
+        this.terminateSeCommunication();
```

Comment:
In my mind terminateSeCommunication() is also called when (this instanceof ObservableReader) && (((ObservableReader) this).countObservers() <> 0))

Reply:
In this case, it is necessary to add terminateSeCommunication as a member of the ObservableReader interface. Is this a good idea?